src/main/scala/mesosphere/marathon/api/RestResource.scala

```diff
@@ -55,22 +54,32 @@ trait RestResource {
   //scalastyle:off
   /**
     * Checks if the implicit validator yields a valid result.
-    * @param t object to validate
+    *
+    * @param ft           function that returns the object to validate
     * @param description optional description which might be injected into the failure message
-    * @param fn function to execute after successful validation
-    * @param validator validator to use
+    * @param fn          function to execute after successful validation
+    * @param validator   validator to use
     * @tparam T type of object
     * @return returns a 422 response if there is a failure due to validation. Executes fn function if successful.
     */
-  protected def withValid[T](t: T, description: Option[String] = None)(fn: T => Response)(implicit validator: Validator[T]): Response = {
+  protected def withValid[T](ft: => T, description: Option[String] = None)(fn: T => Response)(implicit validator: Validator[T]): Response = {
     //scalastyle:on
-    validator(t) match {
-      case f: Failure =>
-        val entity = Json.toJson(description.map(f.withDescription).getOrElse(f)).toString
+    try {
+      val t: T = ft
+      validator(t) match {
+        case f: Failure =>
+          val entity = Json.toJson(description.map(f.withDescription).getOrElse(f)).toString
+          //scalastyle:off magic.number
+          Response.status(422).entity(entity).build()
+        //scalastyle:on
+        case Success => fn(t)
+      }
+    } catch {
+      case ValidationFailedException(validator, failure) =>
```

Comment:
The reason for the try/catch is that we now have 2 validation phases and the results of the first one land here by means of the ValidationFailedException. This has nothing to do with the description issue, with AFAIU is unchanged.